### ISSUE TITLE
Add OEM badge on Product page

### ISSUE THREAD

**Issue by erinemay** [iFixit/react-commerce#797]
https://www.figma.com/file/jRG4EyQXRDMGfyLqut08KP/iFixit---Working-files?node-id=4849%3A48376

**Comment by dhmacs** [comment]
This is the mockup:

<img width="439" alt="Screenshot 2022-10-07 at 12 01 41" src="https://user-images.githubusercontent.com/4640135/194528483-77659e67-24b5-4d16-9a6c-25944137f2d5.png">

I should display the partner logo, but what I have in the metafield value is something like: `Genuine Samsung Galaxy Part`

I could parse the string to detect what's the partner (e.g. "Samsung" in the previous example), but this seems not to be bullet proof as we might have unintended brand name collisions.

I also need a list of all available partner, to do the mapping with the partner logo.

cc @erinemay @sterlinghirsh

**Comment by sterlinghirsh** [comment]
We'll try to get some images, but until we get them, maybe we can just show the text and the link (and then in the future we can do that for any values without matching images).

**Comment by dhmacs** [comment]
Getting the images is not an issue, Mattia has already created a bunch of images for partnerships. The issue is how to do the mapping, and that issue applies also to the "learn more" link.

Things needed to address this issue:

1. How do we map"OEM Partnership" metafield with link and image? Are we going to introduce new metafields for this or am I supposed to try to infer the partner from the "OEM Partnership" metafield?
2. What's the complete list of partners?
3. Where should "Learn more" point to?

**Comment by sterlinghirsh** [comment]
Waiting on an answer from Jeff on this one.

**Comment by danielbeardsley** [comment]
We can add a new metafield that will be the more stable `oem_partnership` code, i.e. one of these:

![image](https://user-images.githubusercontent.com/26855/197072722-c51888d1-ad75-494e-94f0-d2585d8ce9d8.png)

**Comment by federicobadini** [comment]
A couple of additional questions on this:

I have seen that the linked PR on `ifixit/ifixit` is going to provide a field called `oem_partnership_json` that will include both a text and a code.

The badge mockup however is showing both a label and a link:

<img width="758" alt="Screenshot 2022-10-24 at 15 43 11" src="https://user-images.githubusercontent.com/7805759/197540944-059da782-0c18-4a17-bd8d-811ccbb5510b.png">

The label can probably be assembled by adding a prefix to the `oem_partnership_json.text`.
The link however seems not to be following a precise url convention: if you visit the [collaborations page](https://www.ifixit.com/collaborations) you can see that most of the linked pages adopts different url conventions.

Just to name a few variations:

- https://www.ifixit.com/Device/Fairphone
- https://www.ifixit.com/collaborations/samsung
- https://www.ifixit.com/google

If we don't have a precise mapping we may need also this info in the json.

Finally, is the list here complete?
I'm asking just to double check that we have already produced all the needed partnership images.
cc @mmarcon91 

> ![image](https://user-images.githubusercontent.com/26855/197072722-c51888d1-ad75-494e-94f0-d2585d8ce9d8.png)

**Comment by danielbeardsley** [comment]
I'm tempted to say we should store all of the links as code in the php app (since most of their pages probably exist there) and then include another `url` attribute in the oem json.

If someone has an authoritative list of urls, I can add them. if not I can dig through our code.

**Comment by danielbeardsley** [comment]
I'm deciding. Expect that the metafield `ifixit.oem_partnership_json` will have this schema:

```
null | {
   text: string;
   code: string; // One of the above values
   url: string | null; // if null, don't make it a link, just a badge.
}
```

**Comment by federicobadini** [comment]
Perfect. I'll go on with the implementation

> Finally, is the list here complete?
> I'm asking just to double check that we have already produced all the needed partnership images.

Have you any update on this too? ☝️

**Comment by danielbeardsley** [comment]
Yes, that list is complete.

**Comment by federicobadini** [comment]
This is still open since #918 only referenced it. 
Do we wanna keep this open to keep track of the brands that still need to be enabled or do we wanna close this and open a separate issue for the follow up?

**Comment by danielbeardsley** [comment]
I'd vote for a separate issue. Though I'm OK with creating an issue only once we've gotten permission.
